bxparks/AceTime
# merge 2.2.3 into master

* 2.2.3 (2023-05-31, TZDB version 2023c)
    * Update `ace_time/testing/*` classes to support splitting the test data
      into 2 separate lists: `transitions` and `samples`.
        * Required to support new `validation_data.json` from AceTimeValidation.
    * Update `AceTimeValidation` test names to `tests/Xxx{Basic,Extended}Test`.
    * Add `ZonedExtra::kAbbrevSize` to define the `char` buffer size needed to
      hold an abbreviation.
    * Change `ZonedExtra::kInvalidMinutes` from public to private.
        * This is an implementation detail.
        * Use `ZonedExtra::isError()` instead.
    * Rename AceTimePython to acetimepy.
    * ZoneProcessor transition cache
        * Save the epoch year, and automatically invalidate and regenerate the
          cache when the `Epoch::currentEpochYear()` is modified.
        * Remove cache invalidation methods which are no longer needed:
            * `ZoneProcessor::resetTransitionCache()`
            * `ZoneProcessorCache::resetZoneProcessors()`
            * `ZoneManager::resetZoneProcessors()`
    * Rename "Converter Epoch" to "Internal Epoch".
        * Change `daysToCurrentEpochFromConverterEpoch()` to
          `daysToCurrentEpochFromInternalEpoch()`.
        * This is an internal implementation detail, exposed only for testing
          purposes.
    * Update supported boards and tiers
        * Add SAMD21 and SAMD51 boards to Tier 1
            * Add Adafruit ItsyBitsy M4 (SAMD51 120MHz ARM Cortex-M4)
            * SAMD21 and SAMD51 boards are back in Tier 1, as long as they use
              the traditional Arduino API instead of the new
              [Arduino-Core](https://github.com/arduino/ArduinoCore-api).
            * Fortunately most third party SAMD21 and SAMD51 boards continue to
              use the traditional Arduino API.
        * Move Teensy 3.2 to Tier 2
            * This board is entering end-of-life.
            * As well, the Teensyduino environment integrates with the Arduino
              IDE and CLI in a way that's different than all other third-party
              Arduino boards. Some of my automation scripts do not work with
              Teensyduino, so it becomes very time consuming to test the Teensy
              boards.
            * All Teensy boards are now in Tier 2 ("Should work but not tested
              often").
